zipline/pipeline/factors/factor.py

```diff
@@ -5,20 +5,22 @@
 from operator import attrgetter
 from numbers import Number
 
-from numpy import inf
+from numpy import inf, where, nanmean, nanstd
```

Comment:
Should we use `nanmean` from our math_utils instead?  It uses bottleneck if it's available.